sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -447,7 +447,7 @@ trait Row extends Serializable {
             if (d1.compareTo(o2.asInstanceOf[java.math.BigDecimal]) != 0) {
               return false
             }
-          case _ => if (o1 != o2) {
+          case _ => if (!o1.equals(o2)) {
```

Comment:
I was told that these 2 are the same in scala... seems they are different for primitive types.

Reply:
I found this: https://alexknvl.com/posts/eq.html